File: Modules/SIE.ZYF/SIE.Web.ZYF/ProductManages/Commands/AddProductManageCommand.js
SIE.defineCommand("SIE.Web.ZYF.ProductManages.Commands.AddProductManageCommand", {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit", iconCls: "icon-AddEntity icon-green" },
    
    showView: function (entity) {
        var model = entity.data;
        var me = this;
        this.view.execute({
            data: model,
            success: function (res) {
                var code = res.Result.Code;
                var qty = res.Result.PurchaseQuantity;
                CRT.Workbench.addPage({
                    entityType: me.view.model,
                    recordId: entity.getId(),   
                    viewGroup: "DetailsView",
                    title: me.getEditViewTitle(entity),
                    params: {
                        Code: code,
                        Qty: qty
                    },
                    isDetail: true,
                })
            }
        }, me.view);
    }
});

File: Modules/SIE.ZYF/SIE.Web.ZYF/ProductManages/Commands/AddProductManageCommand.cs
using SIE.Web.Command;
using SIE.ZYF.ProductManages;

namespace SIE.Web.ZYF.ProductManages.Commands
{
    public class AddProductManageCommand : ViewCommand
    {

        protected override object Excute(ViewArgs args, string scope)
        {
            var productManage = args.Data.ToJsonObject<ProductManage>();
            productManage.Code = RT.Service.Resolve<ProductManageController>().GetCode();
            productManage.PurchaseQuantity = RT.Service.Resolve<ProductManageController>().GetQuantity();
            return productManage;
        }
    }
}
